packages/perseus-editor/src/components/locked-vector-settings.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * LockedVectorSettings is a component that allows the user to edit the
+ * settings of specifically a locked vector on the graph.
+ *
+ * Used in the interactive graph editor's locked figures section.
+ */
+import {vector as kvector} from "@khanacademy/kmath";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import {spacing, color as wbColor} from "@khanacademy/wonder-blocks-tokens";
+import {LabelLarge, LabelMedium} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+import {useState} from "react";
+
+import ColorSelect from "./color-select";
+import CoordinatePairInput from "./coordinate-pair-input";
+import LineSwatch from "./line-swatch";
+import LockedFigureSettingsAccordion from "./locked-figure-settings-accordion";
+import LockedFigureSettingsActions from "./locked-figure-settings-actions";
+
+import type {AccordionProps} from "./locked-figure-settings";
+import type {
+    Coord,
+    LockedFigure,
+    LockedFigureColor,
+    LockedVectorType,
+} from "@khanacademy/perseus";
+
+const lengthErrorMessage = "The vector cannot have length 0.";
+
+export type Props = LockedVectorType &
+    AccordionProps & {
+        /**
+         * Called when the delete button is pressed.
+         */
+        onRemove: () => void;
+        /**
+         * Called when the props (points, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedFigure>) => void;
+    };
+
+const LockedVectorSettings = (props: Props) => {
+    const [tailCoordExpanded, setTailCoordExpanded] = useState(false);
+    const [tipCoordExpanded, setTipCoordExpanded] = useState(false);
```

Comment:
Does the accordion still work if you remove these states? [WB AccordionSection should work as a controlled component](https://khan.github.io/wonder-blocks/?path=/story/packages-accordion-accordionsection--controlled).

Reply:
Done. Thanks!